examples/hitl/rearrange_v2/rearrange_v2.py

```diff
@@ -349,31 +358,29 @@ def _update_grasping_and_set_act_hints(self, user_index: int):
             reach_pos=None,
         )
 
-    def _get_gui_controlled_agent_index(self, user_index) -> int:
+    def _get_gui_controlled_agent_index(self, user_index):
         return self._gui_agent_controllers[user_index]._agent_idx
 
     def _get_controls_text(self, user_index: int):
-        if self._paused:
-            return "Session ended."
-
-        if not self._user_data[user_index].show_gui_text:
-            return ""
-
         controls_str: str = ""
-        controls_str += "H: Toggle help\n"
-        controls_str += "Look: Middle click (drag), I, K\n"
-        controls_str += "Walk: W, S\n"
-        controls_str += "Turn: A, D\n"
-        controls_str += "Finish episode: Zero (0)\n"
-        controls_str += "Open/close: Double-click\n"
-        controls_str += "Pick object: Double-click\n"
-        controls_str += "Place object: Right click (hold)\n"
+        if self._user_data[user_index].show_gui_text:
+            controls_str += "H: Toggle help\n"
+            controls_str += "Look: Middle click (drag), I, K\n"
+            controls_str += "Walk: W, S\n"
+            controls_str += "Turn: A, D\n"
+            controls_str += "Finish episode: Zero (0)\n"
+            controls_str += "Open/close: Double-click\n"
+            controls_str += "Pick object: Double-click\n"
+            controls_str += "Place object: Right click (hold)\n"
+
+        client_helper = self._app_service.remote_client_state._client_helper
+        idle_time = client_helper.get_idle_time(user_index)
+        if idle_time > 10:
+            controls_str += f"Idle time {idle_time}s\n"
+
         return controls_str
 
     def _get_status_text(self, user_index: int):
```

Comment:
Ooh! Lovely. I can pass `current_high_level_action` being executed by LLM-agent here.

Reply:
If time permits, I can also add a `text` field to viewports so that its rendered into the picture-in-picture view.